gtsfm/pose_slam/pose_slam.py

```diff
@@ -0,0 +1,89 @@
+"""Pose SLAM initialization module from relative pose priors.
+
+Authors: Akshay Krishnan, Frank Dellaert
+"""
+
+from typing import Dict, List, Optional, Tuple
+
+import dask
+import gtsam
+from dask.delayed import Delayed
+from gtsam import Pose3
+
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.pose_prior import PosePrior
+from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
+
+PRIOR_NOISE_SIGMAS = [0.001, 0.001, 0.001, 0.1, 0.1, 0.1]
+POSE_PRIOR_NOISE = gtsam.noiseModel.Diagonal.Sigmas(PRIOR_NOISE_SIGMAS)
+
+logger = logger_utils.get_logger()
+
+
+class PoseSlam:
+    """Initializes using PoseSLAM given relative pose priors."""
+
+    def run_pose_slam(
+        self,
+        num_images: int,
+        relative_pose_priors: Dict[Tuple[int, int], PosePrior] = {},
+        gt_wTi_list: Optional[List[Optional[Pose3]]] = None,
+    ) -> Tuple[List[Optional[Pose3]], Optional[GtsfmMetricsGroup]]:
+        """Run the translation averaging.
+
+        Args:
+            num_images: number of camera poses.
+            relative_pose_priors: priors on the pose between camera pairs as (i1, i2): i1Ti2.
+            gt_wTi_list: List of ground truth poses (wTi) for computing metrics.
+
+        Returns:
+            List of Optional[Pose3],
+            GtsfmMetricsGroup of 1DSfM metrics.
+        """
+        logger.info("[pose slam] Running pose SLAM intilialization")
```

Comment:
I guess we dont need module names here too.

Reply:
Removed